r/tests/testthat/test-dataset.R

```diff
@@ -846,6 +846,38 @@ test_that("mutate() with NULL inputs", {
   )
 })
 
+test_that("empty mutate()", {
+  expect_equal(
+    ds %>%
+      mutate() %>%
+      collect(),
+    ds %>%
+      collect()
+  )
+})
+
+test_that("transmute() with NULL inputs", {
+  expect_equal(
+    ds %>%
+      transmute(int = NULL) %>%
+      collect(),
+    ds %>%
+      select() %>%
```

Comment:
I'm not sure that you need the tests both here and in test-dplyr-mutate.R--they're using the same R code. 

Also see ARROW-11328, which suggests `select()` doesn't do what is expected for Datasets, so this test would mean that both are wrong here. (It's a separate problem, about the C++ library and not the R NSE code, so I wouldn't solve it here.)

Reply:
> I'm not sure that you need the tests both here and in test-dplyr-mutate.R--they're using the same R code.

I recognized that the tests currently exercise the same code but I included them both because it seemed safer not to assume that they always would in the future. Do you prefer I remove the duplicate tests?

Reply:
> Also see [ARROW-11328](https://issues.apache.org/jira/browse/ARROW-11328), which suggests `select()` doesn't do what is expected for Datasets, so this test would mean that both are wrong here. (It's a separate problem, about the C++ library and not the R NSE code, so I wouldn't solve it here.)

That behavior seems to have been fixed in 3.0.0

Reply:
It's a fair point. My bigger concern is that I think they're asserting wrong results because of ARROW-11328, so there's not much value at this time. But it's fine.